### PR TITLE
[checklist] Upgrade wording for D1 milestone

### LINKED ISSUE #3679: [checklist] changing wording of D1 for checklist
In reviewing the D1 terminology on our [checklist](https://docs.opentitan.org/doc/project/checklist/), we thought it was too murky and discouraging designers from converting from D0 to D1 in a manner that was consistent with the original intention. This update hopefully makes it a little more clear, but there is still a challenge to create a universal definition that doesn’t require a *lot* of work in the D0 timeline. Some designs like UART could get “mainline functionality” completed in much less percentage time than say OTP or KMAC. I think that is just the nature of some designs. Regardless, I think our intention here is to say D1 roughly represents the design is “mostly specified, most of the primary functionality CSRs are defined, physical considerations have been considered, and enough functionality has been put in place for a basic verification test. Below is the suggestion before and after for 5 items. I would do a PR but I'm still having machine issues. Putting this here so we can discuss in next Silicon meeting.

**BEFORE**

SPEC_COMPLETE
- Spec complete, feature set finalized.

CSR_DEFINED
- CSRs defined and generated.

MEM_INSTANCED_80
- 80% of expected memories instantiated, using behavioral RAMs.

FUNC_IMPLEMENTED
- Main functional path implemented.

IP_INSTANTIABLE
- Unit is able to be instantiated and bound in top level RTL. The unit must not break top level functionality such as propagating X through TL-UL interface, continuously asserting the interrupts, or creating undesired TL-UL transactions.

**AFTER**

SPEC_COMPLETE
- Specification mostly (90%) complete, all features are defined. Specification is submitted into the repo as a markdown document. It is acceptable to make changes for further clarification or more details after the D1 stage.

CSR_DEFINED
- The CSRs required to implement the primary programming model are defined. The .hjson file defining the CSRs is checked into the repo. It is acceptable to add or modify registers during the D2 stage in order to complete implementation.

PHYSICAL_MACROS_DEFINED_80
- All expected memories identified, representative macros instantiated. All other physical elements (analog components, pads, etc) are identified and represented with a behavioral model. It is acceptable to make changes to these physical macros after the D1 stage as long as they do not have a large impact on the expected resulting area (roughly “80% accurate”).

FUNC_IMPLEMENTED
- Mainline functional path is implemented to allow for a basic functionality test by verification. “Feature complete” is the target for D2 status.

IP_INSTANTIABLE
- Unit is able to be instantiated and bound in top level RTL. The design must compile and elaborate cleanly without errors. The unit must not break top level functionality such as propagating X through TL-UL interface, continuously asserting the interrupts, or creating undesired TL-UL transactions.

### PR DESCRIPTION
**Pull request by sjgitty**
Signed-off-by: Scott Johnson <scottdj@google.com>

Fixes #3679 

Upgrade wording for D1 to specify more succinctly the goals of the milestone. If we're OK with this wording I can add the file modifications to the changing checklist item (`MEM_INSTANCED_80` --> `PHYSICAL_MACROS_DEFINED_80`).

### CODE REVIEW

File: doc/project/checklist.md
Comment (imphil) [inline doc/project/checklist.md:15]:
```suggestion
Specification is submitted into the repository as a Markdown document.
```

File: doc/project/checklist.md
Comment (imphil) [inline doc/project/checklist.md:21]:
```suggestion
The Hjson file defining the CSRs is checked into the repository.
```

File: doc/project/checklist.md
Comment (imphil) [inline doc/project/checklist.md:40]:
```suggestion
All expected memories identified, representative macros instantiated.
```

It it sufficient to use the appropriate primitives, e.g. `prim_ram_1p`, or does "macros instantiated" mean I should coordinate with e.g. Michael to get a customized SRAM macro from the memory generator?

File: doc/project/checklist.md
Comment (imphil) [inline doc/project/checklist.md:41]:
```suggestion
All other physical elements (analog components, pads, etc.) are identified and represented with a behavioral model.
```

**Review by imphil — APPROVED** [review]
Thanks Scott!

File: doc/project/checklist.md
Comment (sjgitty) [reply doc/project/checklist.md:40]: The former. Putting down `prim_ram_1p` in this example is sufficient.

**Review by msfschaffner — APPROVED** [review]
LGTM (looks like you have to squash and sign-off the commits again).

**Comment by imphil** [comment]
@sjgitty let me know if I should do it (and this time I'll tell you the trick) :-)

**Comment by sjgitty** [comment]
> @sjgitty let me know if I should do it (and this time I'll tell you the trick) :-)

Teach to fish. I need to squash and sign these, then add the mods to the other checklists after #3772 lands.

**Comment by imphil** [comment]
```
# Use the hub extension to check out the PR (overloads the "git" command)
git checkout https://github.com/lowRISC/opentitan/pull/3790

# Or _alternatively_, use the new official GitHub CLI (gh)
gh pr checkout https://github.com/lowRISC/opentitan/pull/3790

git rebase -i upstream/master
# "fixup" the last two commits into the first one

# push to update the PR
git push -f
```

**Comment by rswarbrick** [comment]
Following up from Philipp's example, if you don't have any magic GitHub wrappers, you can also do it with "vanilla git" like this. I'm assuming that you've got an `origin` remote pointing at `lowRISC/opentitan` and an `sjgtty` remote pointing at `sjgtty/opentitan`.
```
git fetch origin pull/3790/head
git checkout -b 3790 FETCH_HEAD
git rebase -i origin/master
git push -f sjgtty HEAD:d1wording
```
It's even less painful if you've still got your original branch lying around. Suppose you did your work on a branch called `d1wording` before. Then you can probably just do something like this:
```
git checkout d1wording
git rebase -i origin/master
git push -f sjgtty
```

**Comment by sjgitty** [comment]
None of these worked, alas, which likely means that I've got an old lingering .git/config that is not conformant with the recommended way. Instead I have just redone the two tiny word changes and force pushed them here. So I'm up to date despite the ill-advised method to get there.

I will not update the checklist items to indicate the new name for `MEM_INSTANCED_80` --> `PHYSICAL_MACROS_DEFINED_80`.

File: hw/ip/keymgr/doc/checklist.md
Comment (sjgitty) [inline hw/ip/keymgr/doc/checklist.md:13]: Went ahead and added these missing lines.

**Comment by sjgitty** [comment]
upgraded with the checklist name fixes. @eunchan I ended up changing your checklist the same way @msfschaffner had changed the others, I think there was a TOCTOU condition there. LMK if you're still thumbs up @msfschaffner @imphil

**Review by msfschaffner — APPROVED** [review]
LGTM

**Comment by sriyerg** [comment]
LGTM!

**Review by sriyerg — APPROVED** [review]

**Pull request merged**
